src/components/SearchBar/SearchBar.hook.ts

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect, type ChangeEvent, type KeyboardEvent } from 'react';
+
+import { DEBOUNCE_DELAY } from './SearchBar.constants';
+
+import type { SearchBarProps } from './SearchBar.types';
+
+export const useSearchBar = (props: SearchBarProps) => {
+  const {
+    value: controlledValue,
+    defaultValue,
+    onSearch,
+    debounceDelay = DEBOUNCE_DELAY,
+    onChange,
+  } = props;
+
+  const isControlled = controlledValue !== undefined;
+  const [internalValue, setInternalValue] = useState(defaultValue || '');
+
+  // Sync internal state with controlled value
+  useEffect(() => {
+    if (isControlled) {
+      setInternalValue(controlledValue);
+    }
+  }, [controlledValue, isControlled]);
+
+  // Debounce logic
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      // Only trigger search if not controlled or if explicit requirement
+      // Usually debounce is for auto-search on type.
+      // Requirement says "debounce... onSearch seguro".
+      // If we want auto-search on type:
+      if (onSearch && internalValue) {
+        // onSearch(internalValue); // Uncomment if auto-search is desired
+      }
+    }, debounceDelay);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [internalValue, debounceDelay, onSearch]);
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const newValue = e.target.value;
+
+    if (!isControlled) {
+      setInternalValue(newValue);
+    }
+
+    if (onChange) {
+      onChange(e);
+    }
+  };
+
+  const handleKeyDown = (
+    e: KeyboardEvent<HTMLDivElement | HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    if (e.key === 'Enter') {
```

Comment:
When the component is controlled, onSearch currently uses internalValue. If controlledValue is provided by the parent, internalValue may not reflect the latest controlled value (or may be empty). Use a single derived source-of-truth when calling onSearch so controlled usage works reliably.

**Sugerencia:**
```typescript
Create a currentValue const and use it for all search triggers:
const currentValue = isControlled ? (controlledValue ?? '') : internalValue;
... onSearch(currentValue);
This ensures onSearch sends the actual current value in both controlled and uncontrolled modes.
```